types/strophe.js/muc.d.ts

```diff
@@ -0,0 +1,854 @@
+// Type definitions for strophe.muc.js v1.2.0
+// Project: https://github.com/metajack/strophejs-plugins
+// Definitions by: Ilia Choly <https://github.com/icholy/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { Strophe } from "strophe.js";
+
+declare module "strophe.js" {
+  namespace Strophe {
+      interface Connection {
+      muc: MUC.Plugin;
+    }
+
+    namespace MUC {
+
+      interface Plugin {
+
+        /**
+         * Initialize the MUC plugin. Sets the correct connection object and
+         * extends the namesace.
+         *
+         * @param conn - the connection instance.
+         */
+        init(conn: Connection): void;
+
+        /**
+         * Join a multi-user chat room
+         *
+         * @param room - The multi-user chat room to join.
+         * @param nick - The nickname to use in the chat room. Optional
+         * @param msg_handler_cb - The function call to handle messages from the specified chat room.
+         * @param pres_handler_cb - The function call back to handle presence in the chat room.
+         * @param roster_cb - The function call to handle roster info in the chat room
+         * @param password - The optional password to use. (password protected rooms only)
+         * @param history_attrs - Optional attributes for retrieving history
+         * @param extended_presence - Optional XML for extending presence
+         */
+        join(
+          room:               string,
+          nick:               string,
+          msg_handler_cb:     (stanza: Element, room: XmppRoom) => boolean,
+          pres_handler_cb:    (stanza: Element, room: XmppRoom) => boolean,
+          roster_cb:          (occupants: OccupantMap, room: XmppRoom) => boolean,
+          password?:          string,
+          history_attrs?:     any,
+          extended_presence?: Element
+        ): void;
+
+
+        /**
+         * Leave a multi-user chat room
+         *
+         * @param room - The multi-user chat room to leave.
+         * @param nick - The nick name used in the room.
+         * @param handler_cb - Optional function to handle the successful leave.
+         * @param exit_msg - optional exit message.
+         * @return iqid - The unique id for the room leave.*
+         */
+        leave(
+          room:        string,
+          nick:        string,
+          handler_cb?: Function,
```

Comment:
As discussed [here](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes), try not to use the type `Function`.